microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -299,7 +304,7 @@ def download(self) -> PurePath:
             self._git_ref = git.get_tag(
                 self.asset_path, filter_=r"^v.*"  # starts w 'v'
             )
-            git.checkout(self._git_ref, cwd=self.asset_path)
+            git.checkout(self._git_ref, cwd=self.asset_path, force_run=False)
```

Comment:
What's the reason to set it to false? Git operations are not to get information, it should always run to avoid confusing.

Reply:
Yeah you're right, this will break if you check out a branch, check out another branch, then try to check out the first branch again.

I'm trying to work around an issue with the Git tool. The Git tool creates a branch based on the *run id* if you don't give it a branch name, so will throw if you try to checkout any branches twice on a node without providing some branch name. You can't just pass the branch name as 'branch name' since it uses -b to create a new branch.

Maybe it makes more send to add an option to not try to create a temporary branch?

Reply:
You can specify the `checkout_branch` to avoid generate a local name for branch.

Reply:
This is the problematic code:
```
        result = self.run(
            f"checkout {ref}",
            force_run=True,
            cwd=cwd,
            no_info_log=True,
            no_error_log=True,
        )

        result = self.run(
            f"checkout -b {checkout_branch}",
            force_run=True,
            cwd=cwd,
            no_info_log=True,
            no_error_log=True,
        )
```

Reply:
Oh duh I can just delete the temp branch if it exists before creating it again.

Reply:
fixed